### PR TITLE
Rails version upgraded to 7.0.7

### PR DESCRIPTION
**Pull request by akhilgkrishnan**
Rails recently released the [Rails 7.0.7](https://rubyonrails.org/2023/8/10/Rails-7-0-7-has-been-released).

This PR updates the rails version to 7.0.7

### CODE REVIEW

**Comment by akhilgkrishnan** [comment]
@bdmac Could you please review this PR

File: Gemfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 ruby "~> 3.2.0"
 
 # Bundle edge Rails instead: gem "rails", github: "rails/rails", branch: "main"
-gem "rails", "~> 7.0"
+gem "rails", "~> 7.0.7"
```
Comment (joshaber) [inline Gemfile:7]: Could we let this float and just `bundle update` to bump the lock file? That way we don't need to update this for every Rails update.

**Review by joshaber — COMMENTED** [review]
Just the one question, thanks for doing this @akhilgkrishnan!

File: Gemfile (hunk 1 above)
Comment (akhilgkrishnan) [reply Gemfile:7]: I'm agreeing with you. But it's better if we could mention the exact versions for major dependencies in the gemfile.

**Review by joshaber — APPROVED** [review]

**Pull request merged**